### PR TITLE
[CSL-1531] Recovery ~ too much blocks

### PR DESCRIPTION
**Pull request by volhovM**

### CODE REVIEW

File: src/Pos/Block/Network/Retrieval.hs [hunk 1]
```diff
@@ -122,9 +123,10 @@ retrievalWorkerImpl sendActions =
         updateRecoveryHeader nodeId header
         endedRecoveryVar <- newEmptyMVar
         let cont (headers :: NewestFirst NE (BlockHeader ssc)) =
-                let firstHeader = headers ^. _Wrapped . _neLast
+                let oldestHeader = headers ^. _Wrapped . _neLast
```
Comment (int-index) [inline src/Pos/Block/Network/Retrieval.hs]: Can you avoid `_Wrapped` please? The issue is that if the type changes to `OldestFirst`, the logic will break, but the code will compile. There's a `_NewestFirst` iso for this.

Comment (volhovM) [inline src/Pos/Block/Network/Retrieval.hs]: Alright.

File: core/constants.yaml [hunk 2]
```diff
@@ -31,8 +31,8 @@ dev: &dev
   kademliaDumpInterval: 4
   neighboursSendThreshold: 2
   networkDiameter: 3
-  networkConnectionTimeout: 2000
-  networkReceiveTimeout: 5000
+  networkConnectionTimeout: 2000 # 2 sec
+  networkReceiveTimeout: 20000 # 20 sec
```
Comment (int-index) [inline core/constants.yaml]: Why did you increase the timeout from 5s to 20s? ot baldy?

File: src/Pos/Block/Network/Logic.hs [hunk 3]
```diff
@@ -270,16 +270,19 @@ requestHeaders
...
     inRecovery <- recoveryInProgress
     logDebug $ sformat ("requestHeaders: inRecovery = "%shown) inRecovery
     flip (maybe onNothing) mHeaders $ \(MsgHeaders headers) -> do
         logDebug $ sformat
-            ("requestHeaders: withConnection: received "%listJson%
-             " from nodeId "%build%" of total size "%builder)
-            (map headerHash headers) nodeId (unitBuilder $ biSize headers)
+            ("requestHeaders: received "%int%" headers of total size "%builder%
+             " from nodeId "%build%": "%listJson)
+            (headers ^. _Wrapped . to NE.length)
```
Comment (int-index) [inline src/Pos/Block/Network/Logic.hs]: Minor: I think you could've used `Foldable.length` here and avoid lenses/unwrapping.

File: src/Pos/Block/Network/Retrieval.hs [hunk 4]
```diff
@@ -122,9 +123,10 @@ retrievalWorkerImpl sendActions =
...
+                    newestHeader = headers ^. _Wrapped . _neHead
                 in handleCHsValid sendActions endedRecoveryVar nodeId
-                                  firstHeader (headerHash header)
+                                  oldestHeader (headerHash newestHeader)
```
Comment (int-index) [inline src/Pos/Block/Network/Retrieval.hs]: :+1:

**Pull request merged**